### ISSUE TITLE
installing with `go get -u github.com/linuxkit/linuxkit/src/cmd/linuxkit` fails with "... not enough arguments in call to cli.ContainerCreate ..."

### ISSUE THREAD

**Issue by jf** [linuxkit/linuxkit#3608]
**Description**
 
I previously encountered #3602 trying to install linuxkit. But it seems I've now "progressed" to the following:

```
root@/home# go get -u github.com/linuxkit/linuxkit/src/cmd/linuxkit
# github.com/linuxkit/linuxkit/src/cmd/linuxkit/docker
/root/go/pkg/mod/github.com/linuxkit/linuxkit/src/cmd/linuxkit@v0.0.0-20210215175756-1f93eab506c9/docker/cmd.go:66:38: not enough arguments in call to cli.ContainerCreate
        have (context.Context, *container.Config, nil, nil, string)
        want (context.Context, *container.Config, *container.HostConfig, *network.NetworkingConfig, *v1.Platform, string)
# github.com/radu-matei/azure-vhd-utils/upload/metadata
/root/go/pkg/mod/github.com/radu-matei/azure-vhd-utils@v0.0.0-20170531165126-e52754d5569d/upload/metadata/metaData.go:95:32: blobClient.GetBlobMetadata undefined (type storage.BlobStorageClient has no field or method GetBlobMetadata)
# github.com/radu-matei/azure-vhd-utils/upload
/root/go/pkg/mod/github.com/radu-matei/azure-vhd-utils@v0.0.0-20170531165126-e52754d5569d/upload/upload.go:89:34: cxt.BlobServiceClient.PutPage undefined (type storage.BlobStorageClient has no field or method PutPage)
/root/go/pkg/mod/github.com/radu-matei/azure-vhd-utils@v0.0.0-20170531165126-e52754d5569d/upload/upload.go:93:7: undefined: storage.PageWriteTypeUpdate
root@/home#
```

**Steps to reproduce the issue:**

On Ubuntu 20.04.2 LTS, with go version `go version go1.16 linux/amd64`, run `go get -u github.com/linuxkit/linuxkit/src/cmd/linuxkit`. Go is downloaded and installed from the tarball at https://golang.org/dl/ if that matters (but I don't think it does).

**Comment by stijnopheide** [comment]
I'm having the same problem on ArchLInux with go version go1.16.2 linux/amd64.

**Comment by dave-tucker** [comment]
Ah yes the joys of `go get` and  `go modules`...

`go get -u` in this case attempts to upgrade the dependencies to the latest minor/patch version - it will use `v20.10.5+incompatible`, which includes breaking API changes from `v17.12.0-ce-rc1` that is in `go.mod` so this won't work at all as you've found out.

You can install using `go get github.com/linuxkit/linuxkit/src/cmd/linuxkit` *BUT* note that it will resolve `docker/docker` to `v1.13.1` which it considers to be the latest semver release :facepalm: 

TL;DR don't use `go get`

`go install github.com/linuxkit/linuxkit/src/cmd/linuxkit@latest` will do the right thing, but you'll be missing version/commit information from the binary.

The best option is probably to install a binary release (see Releases tab)

**Comment by jf** [comment]
oh my ... I should give you an award for this. Thanks, @dave-tucker ! I believe those of us who have encountered this issue have been doing so because we've essentially been following the README

**Comment by deitch** [comment]
Awards for Dave! Awards for Dave!

**Comment by dave-tucker** [comment]
I've opened a PR to update the README so we don't lead people in to this trap in future!
![image](https://user-images.githubusercontent.com/3781777/113332989-d01da000-9319-11eb-9721-0696ebf0b97c.png)
